catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/InjectableAttributeRegistry.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.data;
+
+import java.util.Set;
+
+/**
+ * Manages 'injectable' attributes (i.e., attributes that should be injected into certain metacard
+ * types in the system).
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ */
+public interface InjectableAttributeRegistry {
```

Comment:
Instead of making a new registry, I would suggest looking at using the OSGi service registry.  So in this case I would imagine having an `InjectableAttribute` interface that has an attribute and the metacard type target.  You could then register those as services with OSGi.  Then you can use OSGi and Blueprint to handle creating concurrent reference lists and bind/unbinding.